src/eko/runner.py

```diff
@@ -165,36 +166,91 @@ def compute_msbar_mass(theory_card):
         masses: list
             list of msbar masses squared
     """
-    masses = np.full(3, np.inf)
-    nf_active = 3
+    if theory_card["nfref"] is None:
+        raise ValueError(
+            "You need to specify the number of active flavors for alphas \
+            reference value (nfref) when running in MSBAR scheme"
+        )
+    nfa_ref = theory_card["nfref"]
+
+    q2_ref = np.power(theory_card["Qref"], 2)
+    masses = np.concatenate((np.zeros(nfa_ref - 3), np.full(6 - nfa_ref, np.inf)))
     config = {
         "as_ref": theory_card["alphas"],
-        "q2a_ref": np.power(theory_card["Qref"], 2),
+        "q2a_ref": q2_ref,
         "order": theory_card["PTO"],
         "fact_to_ren": theory_card["fact_to_ren_scale_ratio"] ** 2,
+        "method": strong_coupling_mod_ev(theory_card["ModEv"]),
+        "nfref": nfa_ref,
     }
-    for qidx, hq in enumerate("cbt"):
+
+    # First you need to look for the thr around the given as_ref
+    heavy_quarks = quark_names[3:]
+    hq_idxs = np.arange(0, 3)
+    if nfa_ref > 4:
+        heavy_quarks = reversed(heavy_quarks)
+        hq_idxs = reversed(hq_idxs)
+
+    # loop on heavy quarks and compute the msbar masses
+    for q_idx, hq in zip(hq_idxs, heavy_quarks):
```

Comment:
All this part looks a bit complicated to me ... can't we do something easier? I think `ThresholdAtlas` could do the work for us: initialize it with the `Q_h` and then locate `Q_ref` and the masses and check their specific nfs

Reply:
I see this implementation is a bit complex, but I thought that this way it was even possible to give `Q_c` > `Q_b` without loosing consistency.

Reply:
Okay, that would indeed in principle be possible - though I wonder whether we really need to consider that case ...

Reply:
Hi Felix, 
I think that adding another layer with a new ``thr Atlas``, will not simplify much the procedure. Even if you have the ``Qthr`` sorted, if your initial conditions are not in the right patch (ex: you give both ``c`` and ``b`` references above ``b`` scale ) you will need a trick like this to compute ``b`` mass before ``c``.
So to wrap up this PR, this implementation is not "plain", but it's quite general, maybe even more than what will be needed, and (I hope) it allows only solution for consistent conditions. I added and example in the docs to show how it works explicitly.

Reply:
Ok, good.

Maybe let's put somewhere a note that we might want to provide some drawings, e.g. put a `TODO` comment below the example (sometimes drawings can make clear something that is very complicated to explain with words and formulas, and since images are not a problem I would do it... but it's not a priority).